File: src/main/java/de/steklopod/sec02/Lec09FluxFromMono.java
package de.steklopod.sec02;

import de.steklopod.utils.Util;
import reactor.core.publisher.Flux;

public class Lec09FluxFromMono {

    public static void main(String[] args) {

 /*       Mono<String> mono = Mono.just("a");

        Flux<String> flux = Flux.from(mono);

        flux.subscribe(Util.onNext());*/

        Flux.range(1, 10)
                .next() // 1
                .filter(i -> i > 3)
                .subscribe(Util.onNext(), Util.onError(), Util.onComplete());


    }

    private static void doSomething(Flux<String> flux){

    }

}

File: src/main/kotlin/de/steklopod/sec02/Lec09FluxFromMono.kt
package de.steklopod.sec02

import de.steklopod.utils.Util.onComplete
import de.steklopod.utils.Util.onError
import de.steklopod.utils.Util.onNext
import reactor.core.publisher.Flux

object Lec09FluxFromMono {
    @JvmStatic
    fun main(args: Array<String>) {

        /*       Mono<String> mono = Mono.just("a");

        Flux<String> flux = Flux.from(mono);

        flux.subscribe(Util.onNext());*/
        Flux.range(1, 10)
            .next() // 1
            .filter { i: Int -> i > 3 }
            .subscribe(onNext(), onError(), onComplete())
    }

    private fun doSomething(flux: Flux<String>) {}
}
